vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -62,4 +78,23 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
         Float newHeight = (new Float(width) / new Float(imageWidth)) * new Float(imageHeight);
         return new ImageData(newHeight.intValue(), width);
     }
+    
+    
+    /**
+     * Method to return the required attributes of a requested page  
+     * 
+     * @param currPage current page requested from ui
+     * @return map of required attributes
+     */ 
+    @Override
+    public Map<String, Object> getImageListingAttr(int currPage) {
+        long totalImages = imageRepo.count();
+        boolean areImgsPerfPacked = totalImages % pageSize == 0 ? true : false;
+        long totalPossiblePages = totalImages / pageSize;
+        final long totalPages = areImgsPerfPacked ? totalPossiblePages : totalPossiblePages + 1;
+        Pageable sortByRequestedField = PageRequest.of(currPage-1, pageSize, Sort.by(SortByField.CREATION_DATE.toString()));
+        Page<VSImage> requestedImages = imageRepo.findAll(sortByRequestedField);
+        Map<String, Object> attributesMap = new HashMap<String, Object>(){{put("images", requestedImages.getContent()); put("totalpages", totalPages); put("currentpage", currPage); }};
```

Comment:
Just return a list of the requested images. the caller of the method knows which page they requested and you could just have a separate method to get the total number of images or pages. Returning a map makes it very error prone as changing the content of the map can be done without any compiler error, which makes it very error prone. hardcoding the keys makes it even worse.